core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -99,12 +110,24 @@ public int read() throws IOException {
     if (mPosition == mLength) { // at end of file
       return -1;
     }
-    updateStream();
-    int result = mBlockInStream.read();
-    if (result != -1) {
-      mPosition++;
+
+    CountingRetry retry = new CountingRetry(MAX_WORKERS_TO_RETRY);
+    while (true) {
```

Comment:
For `CountingRetry` or in general `Retry`, we typically use them in the following pattern. I know it is largely equivalent from what you have here.  But I think the following way is better due to readability---I feel having the `while(true)` loop and break the loop control inside the catch block can be tricky to reason. and the following pattern as our convention is quite clear which part we retry (i.e. the while loop).

```java
IOException te = null;
while (retry. attemptRetry()) {
  updateStream();
  try {
    ...
  } catch (UnavailableException | DeadlineExceededException | ConnectException e) {
    te = w;
    handleRetryableException(mBlockInStream, e);
    mBlockInStream = null;
  }
}
throw te;
```

Reply:
Done.